lib/puppet/http/resolver/settings.rb

```diff
@@ -1,5 +1,6 @@
 class Puppet::HTTP::Resolver::Settings < Puppet::HTTP::Resolver
-  def resolve(session, name, &block)
-    yield session.create_service(name)
+  def resolve(session, name, ssl_context: nil, &block)
+    service = Puppet::HTTP::Service.create_service(@client, name)
+    return service if check_connection?(session, service, ssl_context: ssl_context)
```

Comment:
If `check_connection?` returns falsey, then ruby implicitly returns `nil`. To make it clearer, I'm a fan of making the falsey case explicit like:

```ruby
check_connection?(session, service, ssl_context: ssl_context) ? service : nil
```